apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Span.java

```diff
@@ -285,8 +285,8 @@ protected void afterEnd() {
             }
             if (!buffered.tryToCompress(this)) {
                 if (parent.bufferedSpan.compareAndSet(buffered, this)) {
-                    this.tracer.endSpan(buffered);
                     parent.decrementReferences();
```

Comment:
Should we store `parent` in a local variable so that we have a stable reference even if the instance variable changes?

Reply:
no, this is recycling and just the order here was wrong. better not to hang on to objects being recycled